### PR TITLE
avro2sft jahhulbert

### PR DESCRIPTION
**Pull request by jahhulbert-ccri**
A few things that we probably still need to do:
* backwards compat testing

For avro generically:
* instead of using bytes for lists/maps do array, map as avro schema items so that its more avro friendly and not an opaque byte array

Also, didn't automatically change ```-``` to ```_``` but we can consider that I guess. These don't easily encode/decode back and forth anyways since this impl doesn't always escape underscores but that would be the proper thing to do like to do with table names :computer:

### CODE REVIEW

File: geomesa-features/geomesa-feature-avro/src/main/scala/org/locationtech/geomesa/features/avro/FieldNameEncoder.scala [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+  def deHexEscape(s: String): String = {
+    val chars = s.toCharArray
+    var idx = 0
+    val sb = new StringBuilder
+    while (idx < chars.length) {
+      if (chars(idx).isLetterOrDigit) {
+        sb.append(chars(idx))
+        idx += 1
+      } else {
+        val hex = chars(idx + 1).toString + chars(idx + 2).toString
```
Comment (elahrvivaz) [inline geomesa-features/geomesa-feature-avro/src/main/scala/org/locationtech/geomesa/features/avro/FieldNameEncoder.scala]: could this run into issues if there were literal underscores in the original string?

File: geomesa-features/geomesa-feature-avro/src/test/scala/org/locationtech/geomesa/features/avro/AvroSimpleFeatureWriterTest.scala [hunk 2]
```diff
@@ -131,6 +133,26 @@ class AvroSimpleFeatureWriterTest extends Specification with Mockito with Abstra
 
       there was one(encoder).writeArrayEnd()
     }
+
+    "use unmangled names when requested" >> {
+      import org.locationtech.geomesa.security._
+      val sf = createSimpleFeature
+
+      val vis = "test&usa"
```
Comment (elahrvivaz) [inline geomesa-features/geomesa-feature-avro/src/test/scala/org/locationtech/geomesa/features/avro/AvroSimpleFeatureWriterTest.scala]: might not want to have this as a visibility in our code...

File: geomesa-tools/README.md [hunk 3]
```diff
@@ -825,6 +827,27 @@ Export queries and statistics logged for raster tables by using the `querystats`
 
 #### Example command:
     geomesa queryrasterstats -u username -p password -t somerastertable -num 10
+
+### sft2avro
```
Comment (elahrvivaz) [inline geomesa-tools/README.md]: maybe we could rename this to something like 'generateAvroSchema' so we can get away from the x2x syntax that we tried to remove from the converters.

File: geomesa-features/geomesa-feature-avro/src/test/scala/org/locationtech/geomesa/features/avro/AvroSimpleFeatureWriterTest.scala (hunk 2 above)
Comment (jahhulbert-ccri) [inline geomesa-features/geomesa-feature-avro/src/test/scala/org/locationtech/geomesa/features/avro/AvroSimpleFeatureWriterTest.scala]: yup

File: geomesa-features/geomesa-feature-avro/src/main/scala/org/locationtech/geomesa/features/avro/FieldNameEncoder.scala (hunk 1 above)
Comment (jahhulbert-ccri) [inline geomesa-features/geomesa-feature-avro/src/main/scala/org/locationtech/geomesa/features/avro/FieldNameEncoder.scala]: totally. i'll just escape them too like we do in the table names

File: geomesa-tools/README.md (hunk 3 above)
Comment (jahhulbert-ccri) [inline geomesa-tools/README.md]: cool

**Comment by elahrvivaz** [comment]
e49df1d9696edaca15cc2697665beb2bb711cab1

**Pull request closed without merging**